turbopack/crates/turbopack-ecmascript/src/analyzer/well_known.rs

```diff
@@ -228,67 +228,100 @@ pub fn path_resolve(cwd: JsValue, mut args: Vec<JsValue>) -> JsValue {
         return args.into_iter().next().unwrap();
     }
 
-    // path.resolve stops at the string starting with `/`
-    for (idx, arg) in args.iter().enumerate().rev() {
-        if idx != 0 {
-            if let Some(str) = arg.as_str() {
-                if str.starts_with('/') {
-                    return path_resolve(cwd, args.drain(idx..).collect());
+    let args_concat = JsValue::concat(args);
+    let binding = args_concat.expand_alternatives();
+    let args_alt = match binding.as_ref() {
+        JsValue::Concat(_, args) => vec![args],
+        JsValue::Alternatives { values, .. } => values
+            .iter()
+            .map(|v| {
+                if let JsValue::Concat(_, args) = v {
+                    args
+                } else {
+                    unreachable!()
                 }
-            }
-        }
-    }
+            })
+            .collect(),
+        _ => unreachable!(),
+    };
 
-    let mut results_final = Vec::new();
-    let mut results: Vec<JsValue> = Vec::new();
-    for item in args {
-        if let Some(str) = item.as_str() {
-            for str in str.split('/') {
-                match str {
-                    "" | "." => {
-                        if results_final.is_empty() && results.is_empty() {
-                            results_final.push(str.into());
-                        }
+    let mut args_alt: Vec<_> = args_alt
+        .into_iter()
+        .map(|args| {
+            // path.resolve stops at the first (last) absolute string (starting with `/` or `\\`)
+            for (idx, arg) in args.iter().enumerate().rev() {
+                if let Some(str) = arg.as_str() {
+                    if idx != 0 && str.starts_with('/') {
+                        return path_resolve(cwd.clone(), args.clone().drain(idx..).collect());
                     }
-                    ".." => {
-                        if results.pop().is_none() {
-                            results_final.push("..".into());
+                    if let Some(str) = str.strip_prefix("\\\\") {
+                        let str = format!("/{}", str);
+                        let args = std::iter::once(str.into())
+                            .chain(args.clone().drain((idx + 1)..))
+                            .collect();
+                        return path_resolve(cwd.clone(), args);
+                    }
```

Comment:
That doesn't make sense to me. A string starting with `/` marks the root on linux, but a string starting with `\` doesn't mark the root on windows. It would need a drive letter.

Reply:
Network shares are specified via `\\HOSTNAME_OR_IP\subfolder`
And this is also how `path.resolve` behaves:
```
> path.win32.resolve("prefix", "\\", "foo")
'\\foo'
```